core/src/main/scala/org/apache/spark/util/UninterruptibleThread.scala

```diff
@@ -35,8 +35,38 @@ private[spark] class UninterruptibleThread(
     this(null, name)
   }
 
+  class UninterruptibleLock {
+    def awaitInterrupt(): Boolean = synchronized {
```

Comment:
It's not very intuitive to know the relationship between the boolean return value and the function name. So returning true means keep waiting? shall we name the function `shouldAwaitInterrupt`?

Reply:
@cloud-fan did more refactoring, please take a look